### ISSUE TITLE
Unable to build latest container -- Error in /musicbrainz-server/script/compile_resources.sh

### ISSUE THREAD

**Issue by isuftin** [metabrainz/musicbrainz-docker#14]
Running: `docker-compose run --rm musicbrainz /createdb.sh -fetch`

Error:

```
Step 22 : RUN /musicbrainz-server/script/compile_resources.sh
 ---> Running in f6099aa17eda
Use of uninitialized value $ENV{"DB_ENV_POSTGRES_USER"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_ENV_POSTGRES_PASSWORD"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_PORT_5432_TCP_ADDR"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_PORT_5432_TCP_PORT"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_ENV_POSTGRES_USER"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_ENV_POSTGRES_PASSWORD"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_PORT_5432_TCP_ADDR"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_PORT_5432_TCP_PORT"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_ENV_POSTGRES_USER"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_ENV_POSTGRES_PASSWORD"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_PORT_5432_TCP_ADDR"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Use of uninitialized value $ENV{"DB_PORT_5432_TCP_PORT"} in string at /musicbrainz-server/admin/../lib/DBDefs.pm line 51.
Attribute (port) does not pass the type constraint because: Validation failed for 'Int' with value "" at constructor MusicBrainz::Server::Database::new (defined at /musicbrainz-server/admin/../lib/MusicBrainz/Server/Database.pm line 53) line 52
        MusicBrainz::Server::Database::new('MusicBrainz::Server::Database', 'HASH(0x3c62b00)') called at /musicbrainz-server/admin/../lib/MusicBrainz/Server/DatabaseConnectionFactory.pm line 19
        MusicBrainz::Server::DatabaseConnectionFactory::register_databases('MusicBrainz::Server::DatabaseConnectionFactory', 'READWRITE', 'HASH(0x3c55f00)', 'TEST', 'HASH(0x3c560b0)', 'READONLY', 'HASH(0x3c62b00)', 'SYSTEM', 'HASH(0x3c627a0)') called at /musicbrainz-server/admin/../lib/DBDefs.pm line 51
        require DBDefs.pm at admin/ShowDBDefs line 38
        main::BEGIN at /musicbrainz-server/admin/../lib/DBDefs.pm line 0
        eval {...} at /musicbrainz-server/admin/../lib/DBDefs.pm line 0
Compilation failed in require at admin/ShowDBDefs line 38.
BEGIN failed--compilation aborted at admin/ShowDBDefs line 38.
/musicbrainz-server/root/static/gulpfile.js:4
  throw new Error('error: DEVELOPMENT_SERVER should be set to either 0 or 1');
  ^

Error: error: DEVELOPMENT_SERVER should be set to either 0 or 1
    at Object.<anonymous> (gulpfile.js:2:9)
    at Module._compile (module.js:413:34)
    at loader (/musicbrainz-server/node_modules/babel-register/lib/node.js:130:5)
    at Object.require.extensions.(anonymous function) [as .js] (/musicbrainz-server/node_modules/babel-register/lib/node.js:140:7)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/musicbrainz-server/gulpfile.js:2:1)
    at Module._compile (module.js:413:34)
```

docker-compose.yml:

```
postgresqldata:
  build: data-dockerfile
  restart: always
  volumes:
   - /datadump:/media/dbdump:rw

postgresql:
  build: postgres-dockerfile
  restart: always
  env_file:
   - ./postgres-dockerfile/postgres.env
  expose:
   - "5432"

musicbrainz:
  build: musicbrainz-dockerfile
  ports:
   - "5000:5000"
  volumes_from:
   - postgresqldata
  restart: always
  environment:
   - MUSICBRAINZ_USE_PROXY=1
   - DEVELOPMENT_SERVER=1
  expose:
   - "5000"
  links:
   - postgresql:db
```

```
$ docker --version
Docker version 1.11.1, build 5604cbe

$ docker-compose --version
docker-compose version 1.7.1, build 0a9ab35
```

DBDEFs.pm has `sub DEVELOPMENT_SERVER { 0 }`

**Comment by orangepeelbeef** [comment]
same problem, its also complaining about the db port though so maybe that's the real culprit.

**Comment by orangepeelbeef** [comment]
I was able to fix this by editing musicbrainz-dockerfile/DBDefs.pm 
it looks like he's expecting some environment variables which may or may not work, but i hardcoded credentials in mine and it seems to work.

**Comment by jsturgis** [comment]
Yeah those values should be exposed by docker when it links to the postgresql image. I am not seeing this problem on my instance but I will look into it and try to get this fixed. In the meantime you can do what @orangepeelbeef said and hard code those values.

**Comment by jsturgis** [comment]
@isuftin @orangepeelbeef, if you replace CONTAINER_ID in the command below with your running musicbrainz container id and run it do you see DB_ENV_POSTGRES_USER, DB_ENV_POSTGRES_PASSWORD or DB_PORT_5432?

```
sudo docker exec -ti CONTAINER_ID /usr/bin/printenv | /usr/bin/awk '/^DB_ENV_POSTGRES/ || /^DB_PORT_5432_TCP/'
```

**Comment by isuftin** [comment]
@jsturgis Those env vars are not available in my running Docker container

**Comment by jsturgis** [comment]
did you run ```docker-compose up -d``` before running ```docker-compose run --rm musicbrainz /createdb.sh -fetch```?

**Comment by jsturgis** [comment]
I pushed a fix, let me know if it fixes your issues.

**Comment by orangepeelbeef** [comment]
Step 30 : CMD /start.sh
 ---> Running in fdbd25fc2abe
 ---> 9e180ca1a9e0
Removing intermediate container fdbd25fc2abe
Successfully built 9e180ca1a9e0
WARNING: Image for service musicbrainz was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Creating musicbrainzdocker_postgresqldata_1
Creating musicbrainzdocker_postgresql_1
Creating musicbrainzdocker_musicbrainz_1

ERROR: for musicbrainz  b'driver failed programming external connectivity on endpoint musicbrainzdocker_musicbrainz_1 (bc58cd19c77b621bad43cef3a2eaf4a1d072f527d55e64071768971d73b8bb63): Bind for 0.0.0.0:5000 failed: port is already allocated'
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/compose/cli/main.py", line 58, in main
    command()
  File "/usr/lib/python3.5/site-packages/compose/cli/main.py", line 109, in perform_command
    handler(command, command_options)
  File "/usr/lib/python3.5/site-packages/compose/cli/main.py", line 734, in up
    remove_orphans=remove_orphans)
  File "/usr/lib/python3.5/site-packages/compose/project.py", line 402, in up
    'Encountered errors while bringing up the project.'
compose.project.ProjectError: Encountered errors while bringing up the project.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/docker-compose", line 9, in <module>
    load_entry_point('docker-compose==1.7.1', 'console_scripts', 'docker-compose')()
  File "/usr/lib/python3.5/site-packages/compose/cli/main.py", line 63, in main
    log.error(e.msg)
AttributeError: 'ProjectError' object has no attribute 'msg'

**Comment by orangepeelbeef** [comment]
I believe it is working now.

I did notice however:

2016-05-24 13:23:20 (5.62 MB/s) - '/media/dbdump/.listing' saved [1880]

Removed '/media/dbdump/.listing'.
Rejecting 'mbdump-documentation.tar.bz2'.
Rejecting 'mbdump-documentation.tar.bz2.asc'.
Rejecting 'mbdump-edit.tar.bz2'.
Rejecting 'mbdump-edit.tar.bz2.asc'.
Rejecting 'mbdump-editor.tar.bz2'.
Rejecting 'mbdump-editor.tar.bz2.asc'.

FINISHED --2016-05-24 13:26:42--
Total wall clock time: 3m 22s
Downloaded: 16 files, 2.3G in 3m 21s (11.6 MB/s)
/media/dbdump /musicbrainz-server
mbdump-cdstubs.tar.bz2: OK
mbdump-cover-art-archive.tar.bz2: OK
mbdump-derived.tar.bz2: OK
md5sum: mbdump-documentation.tar.bz2: No such file or directory
mbdump-documentation.tar.bz2: FAILED open or read
md5sum: mbdump-edit.tar.bz2: No such file or directory
mbdump-edit.tar.bz2: FAILED open or read
md5sum: mbdump-editor.tar.bz2: No such file or directory
mbdump-editor.tar.bz2: FAILED open or read
mbdump-stats.tar.bz2: OK
mbdump-wikidocs.tar.bz2: OK
mbdump.tar.bz2: OK
md5sum: WARNING: 3 listed files could not be read
